mesonbuild/programs.py

```diff
@@ -100,16 +100,25 @@ def description(self) -> str:
         '''Human friendly description of the command'''
         return ' '.join(self.command)
 
-    def get_version(self, interpreter: 'Interpreter') -> str:
+    def get_version(self, interpreter: T.Optional['Interpreter'] = None) -> str:
         if not self.cached_version:
             from . import build
             raw_cmd = self.get_command() + ['--version']
-            res = interpreter.run_command_impl(interpreter.current_node, (self, ['--version']),
-                                               {'capture': True, 'check': True, 'env': build.EnvironmentVariables()},
-                                               True)
-            output = res.stdout.strip()
+            if interpreter:
+                res = interpreter.run_command_impl(interpreter.current_node, (self, ['--version']),
+                                                   {'capture': True,
+                                                    'check': True,
+                                                    'env': build.EnvironmentVariables()},
+                                                   True)
+                o, e = res.stdout, res.stderr
+            else:
+                p, o, e = mesonlib.Popen_safe(raw_cmd)
+                if p.returncode != 0:
+                    cmd_str = ' '.join(raw_cmd)
```

Comment:
I would use `mesonlib.join_args()` here.

Reply:
It's just for logging the error, but better use it indeed. Done.